src/blocksparsearray/blockdiagonalarray.jl

```diff
@@ -0,0 +1,30 @@
+# type alias for block-diagonal
+using LinearAlgebra: Diagonal
+
+const BlockDiagonal{T,A,Axes,V<:AbstractVector{A}} = BlockSparseMatrix{
+  T,A,Diagonal{A,V},Axes
+}
+
+function BlockDiagonal(blocks::AbstractVector{<:AbstractMatrix})
+  return BlockSparseArray(
+    Diagonal(blocks), (blockedrange(size.(blocks, 1)), blockedrange(size.(blocks, 2)))
+  )
+end
+
+# SVD implementation
+function eigencopy_oftype(A::BlockDiagonal, T)
+  diag = map(Base.Fix2(eigencopy_oftype, T), LinearAlgebra.diag(blocks(A)))
+  return BlockDiagonal(diag)
+end
+
+function svd(A::BlockDiagonal; kwargs...)
+  return svd!(eigencopy_oftype(A, LinearAlgebra.eigtype(eltype(A))); kwargs...)
+end
+function svd!(A::BlockDiagonal; full::Bool=false, alg::Algorithm=default_svd_alg(A))
+  # TODO: handle full
+  F = map(a -> svd!(a; full, alg), blocks(A).diag)
+  Us = map(Base.Fix2(getproperty, :U), F)
+  Ss = map(Base.Fix2(getproperty, :S), F)
+  Vts = map(Base.Fix2(getproperty, :Vt), F)
+  return SVD(BlockDiagonal(Us), mortar(Ss), BlockDiagonal(Vts))
+end
```

Comment:
Can we merge this implementation with the version `svd!(A::AbstractBlockSparseMatrix)`? I realize this one is a little simpler, but it isn't _that_ much simpler, and I like the idea of having just one implementation since then if we do non-trivial optimizations like multi-thread over blocks we can just do that in one place. Also, I doubt that there is a performance gain for this version vs. the more general version.

Reply:
The most complicated part of `svd!(A::AbstractBlockSparseMatrix)` is `_allocate_svd_output`, which we could specialize for block diagonal matrices.

Reply:
Sure! It was mostly still there because I had it before, but it doesn't change too much anyways.